### DISCUSSION TITLE
Support bei der Freetz-NG Entwicklung und Melden aktueller Probleme

### DISCUSSION THREAD

**Discussion by SvenLuebke** [Freetz-NG/freetz-ng#244] [Allgemein]
Hallo Freetz-NG-Entwickler!

Vorerst möchte ich mich bei allen Leuten bedanken, die bei Freetz(-NG) mitgewirkt haben. Ich habe dadurch viele Vorteile genießen können. Also, vielen Dank an Euch! Macht weiter so!

Mich interessiert außerdem der Status der Freetz-NG-Entwicklung bzgl. Support und Melden von Problemen! Ich benutze Freetz (-NG) mit großer Leidenschaft hauptsächlich zum Bauen von "Packages" für die original Firmware. Wenn ich von einem Package eine neuere Version benötige (z.B. von Subversion), diese aber von Freetz-NG noch nicht unterstützt wird, dann baue ich mir die Funktionalität derzeit zu Hause zusammen (wenn ich es denn hinbekomme). Macht es Sinn zu Forken und dann einen PR zu stellen? Oder seid Ihr an solchen Änderungen nicht interessiert? Ich habe z.B. inzwischen Subversion 1.14.1 unter Freetz-NG laufen (mit gewissen Einschränkungen). Auch möchte ich das vollständige perl auf der Fritzbox laufen haben (für FHEM).

Außerdem hab ich auch noch ein...zwei kleinere Probleme feststellen können (z.B. kann php 5.6.40 derzeit nicht mit openssl 1.1.1 kompiliert werden), ist aber anwählbar. Soll ich für so etwas Issues einstellen oder wollt Ihr damit nicht belästigt werden? Weil ich unbedingt openssl1.1.1 benötige, checke ich gerade, wie ich php8.0.2 zum Laufen bekomme.

Gibt es irgendwo eine Liste, die beschreibt, welche Package-Upgrades mal probiert wurden, aber aus irgendwelchen technischen Gründen nicht umgesetzt werden konnten?

Danke für Eure Infos!

Viele Grüße
Sven

**Comment by fda77** [comment]
Hallo, PRs sind willkommen! Eine kleine Merkliste für bumps: https://freetz-ng.github.io/freetz-ng/SUPPORT.html#package-bump für PHP dürfte da aber noch einiges mehr in der .mk und den Patches zu machen sein.
Neue Packages sind auch prima und unproblematisch da sich da niemand daran stören kann dass etwas nicht mehr funktionert. Pro Package etc am besten einzelne PRs.
Mit Openssl 1.1funktioniert nicht alles. Teilweise sind die Packages so alt und es gibt keine Updates für die neueren Openssl. In diesem Fall hab ich zb für Opendd dies gemacht: https://github.com/Freetz-NG/freetz-ng/commit/557591336f06b800f2513e8959fcfdda0e99e970
Das würde ich für PHP auch vorschlagen, bzw es wäre noch besser wenn es eine neuere Version gäbe die die neueren Openssl unterstützt.
Wenn das für alle Packages mal passt könnte man Openssl 1.0 deaktivieren, es ist eh seit langem nicht mehr supported.
Eigentlich wurden alles Package bump soweit gemacht, bei denen es fehlt hat einfach niemand Zeit/Lust gehabt. zb die neuen getdns/stubby version bräuchte libunound welches es in freetz nicht gibt, und dazu noch cmake was aktuelle auch nicht unterstützt wird

**Comment by SvenLuebke** [comment]
OK, vielen Dank! Sobald ich ein bisschen Zeit habe und meine anderen Probleme gelöst habe, werde ich mal ein paar PRs stellen.
